### DISCUSSION TITLE
The directory has been deleted locally

### DISCUSSION THREAD

**Discussion by rrodrigueznt** [abraunegg/onedrive#2285] [Q&A] [answered]
Hi! Happily running an ```OneDrive v2.4.23``` instance on a ```CentOS Stream 8``` server. After playing a bit to configure OneDrive Business Shared Folders (BSF), I ended up with a unit that refuses to start with the following error message:

```
2023-Jan-07 11:45:55.3767211    Processing .
2023-Jan-07 11:45:55.376769     The directory has not changed
2023-Jan-07 11:45:55.3769462    Processing **folder01**
2023-Jan-07 11:45:55.3770768    The directory has been deleted locally
2023-Jan-07 11:45:55.3771119    Deleting item from OneDrive: **folder01**
2023-Jan-07 11:45:55.4105113    ERROR: An attempt to remove a large volume of data from OneDrive has been detected. Exiting client to preserve data on OneDrive
2023-Jan-07 11:45:55.4106208    ERROR: To delete a large volume of data use --force or increase the config value 'classify_as_big_delete' to a larger value
```

Thank Alex! This software has great security features to avoid doing stupid things!

As far as I can remember, **folder01** is a BSF that never ended to synchronize on this server. I included it in my ```business_shared_folders``` file, but once I started the synchronization, I cancelled it when I realized that it is a big folder already synchronized with a different OneDrive unit. I removed it from ```business_shared_folders```.

So, please, how could I tell OneDrive that forget about **folder01**? Thanks!

**Comment by abraunegg** [comment, marked as answer]
@rrodrigueznt 
> I removed it from business_shared_folders.
> ....
> So, please, how could I tell OneDrive that forget about folder01? Thanks!

So if the folder is already removed from your OneDrive configuration (`business_shared_folders` + any other eg from your `sync_list` query ..):

1. Stop the applicable running process / systemd service
2. Cleanup your local file system
3. Run the application manually using `--resync` to cleanup the local database cache and allow the system to sync. Ensure you specify any applicable `--confdir` option for the instance that was stopped.
4. Restart the applicable systemd service 

This is the only process to cleanup your errant configuration

**Reply by rrodrigueznt** [reply]
Thanks! Still one doubt here. Please, why must I stop any OneDrive running process /systemd service and not only the one affected by the errant configuration?

**Reply by abraunegg** [reply]
Essentially you only need to stop the impacted instance which is giving you the issue

**Reply by rrodrigueznt** [reply]
@abraunegg 

> 1. Stop the applicable running process / systemd service
> 2. Cleanup your local file system

Please, could I use the ```--cleanup-local-files = "true"``` configuration option to remove local copies of folders no longer under synchronisation? For instance, I made a symbolic link between ```business-shared-folders``` and ```sync_list```, must I expect that ```--cleanup-local-files = "true"``` remove all local copies of folders not included in that files? It is not working that way here. Must I clean up manually the local file system? Thanks!

**Reply by abraunegg** [reply]
@rrodrigueznt 
> Please, could I use the --cleanup-local-files = "true" configuration option to remove local copies of folders no longer under synchronisation?

No you cannot ..... as per the man page and documentation:
```
--cleanup-local-files
      Cleanup additional local files when using --download-only. This will remove local data.
```

In your specific situation you *must* clean up your files manually. This advice has been provided above and is explicitly clear.

**Reply by rrodrigueznt** [reply]
@abraunegg 

> No you cannot ..... as per the man page and documentation:
> 
> ```
> --cleanup-local-files
>       Cleanup additional local files when using --download-only. This will remove local data.
> ```

I'm using also ```--download-only``` together with ```--cleanup-local-files```. Here are the uncommented lines of the involved configuration file:

```
sync_dir = "~/OneDrive_USC-BSF"
log_dir = "/var/log/onedrive_usc-bsf/"
download_only = "true"
enable_logging = "true"
no_remote_delete = "true"
cleanup_local_files = "true"
```

**Reply by abraunegg** [reply]
@rrodrigueznt 
Regardless of this - I have advised that your only option in your scenario is to perform a manual cleanup.

Why are you trying to do something different than the advice given? That option will not work for you in your current scenario.

**Reply by rrodrigueznt** [reply]
@abraunegg 

> Why are you trying to do something different than the advice given? That option will not work for you in your current scenario.

Simple! I'm trying to understand the not-easy process behind your/our excellent software!

It is error-prone and time-consuming to clean up dozens of folders manually in the current state of my account. **But of course, I will follow your advice.**

Also, I will try to understand why the combination of ```--download-only="true"``` and ```--cleanup-local-files="true"``` configuration options don't work in my scenario.

Thanks for your work!

**Reply by abraunegg** [reply]
@rrodrigueznt 
> It is error-prone and time-consuming to clean up dozens of folders manually in the current state of my account

From a command prompt - potentially `rm -rf <folder_name>` will assist you .. that should remove that folder locally + any children beneath that folder. But of course - that is a command that can cause major issues if run incorrectly ... so use with caution.

> Also, I will try to understand why the combination of --download-only="true" and --cleanup-local-files="true" configuration options don't work in my scenario.

For the following reasons:
1. The folder already exists locally and you want to remove it - and using `--resync` will not make a difference here
2. You are using Business Shared Folders + Sync List - this means you are forcing a *very* specific sync scope ... and the folders you are wanting to remove - fall outside of the sync scope - thus - are untouched ... as the application has no understanding that folders, outside of your sync scope need to be removed .....

So hang on .. point 2 above and `--cleanup-local-files` appears to be in direct conflict then ... yes this is correct

Why ?

This is because in your *very specific scenario* the sync_list processing is done *before* anything else ... If the folder you want to remove exists in the folders covered by your sync list scope, then the folder, would be cleaned up locally.

The whole issue here is - you had the folder created locally, you then enforced a specific sync scope ... meaning the application will not touch / remove anything else locally .... because it is *out-of_scope* ... thus .. not deleted.

This is why I have been saying, in your specific scenario that manual deletion is the *only* option for you.
